docs/DevGuide/Parallelization.md

```diff
@@ -564,6 +571,78 @@ calls on all array elements occurred in the same order. It is **undefined**
 behavior if the contribute calls are made in different orders on different array
 elements.
 
+# Mutable items in the GlobalCache
+
+Most items in the GlobalCache are constant, and are determined at
+initialization by type aliases called `const_global_cache_tags` as
+described above. However, the GlobalCache can also store mutable
+items. Because of asynchronous execution, care must be taken when mutating
+items in the GlobalCache, as described below.
+
+## 1. Initialization of mutable GlobalCache items
+
+Mutable GlobalCache items are initialized by specifying them via a
+type alias `mutable_global_cache_tags`, which is treated the same way
+as `const_global_cache_tags` for const items.
+
+## 2. Use of mutable GlobalCache items
+
+### 1. Checking if the item is up to date
+
+Because execution is asynchronous, any code that uses a mutable item
+in the GlobalCache must first check whether that item is up to date.
+The information about whether an item is up to date is assumed to be
+stored in the item itself.  For example, a mutable object stored in the
+GlobalCache might have type `std::map<temporal_id,T>` (for some type
+`T`), and then any code that uses the stored object can check whether
+an entry exists for a particular `temporal_id`.
```

Comment:
Emphasize here that when creating a mutable cache item, its readiness should be available in information like the `temporal_id`, for which the readiness is independent of order of mutation. For instance, just checking the size of a vector will probably result in race conditions.